### ISSUE TITLE
Allow postCreateCommand shell to be specified

### ISSUE THREAD

**Issue by rubensa** [microsoft/vscode-remote-release#2201]
Looks like **VSCode** runs this for **postCreateCommand**:
```
docker exec -w /workspaces/<PROJECT_FOLDER> -u <USER> <GENERATED_CONTAINER_ID>  /bin/sh -c <POST_CREATE_COMMAND>
```

I need to change "**/bin/sh -c**" with "**/bin/bash -l -c**" to allow **.profile** an **.bashrc** to be evaluated (see: https://github.com/microsoft/vscode-dev-containers/issues/51 for similar problem).

In my **Dockerfile** I use `SHELL [ "/bin/bash", "-l", "-c" ]` but I can't tell **VSCode** to run that shell for **postCreateCommand**.

I tried with something like:
"postCreateCommand": "/bin/bash -l -c \"command\""`
but this does not work as my command includes a **$(subcommand)** that is not found as **subcommand** is evaluated by **/bin/sh** (not by **/bin/bash**).

Please add a "**postCreateShell**" o similar property to allow to set the desired shell to be run.

**Comment by chrmarti** [comment]
You can set `postCreateCommand` to an array to avoid /bin/sh. Does that work for you?

**Comment by rubensa** [comment]
Yes!  Great!!

Didn't know that behavior (that specifying an array avoids running **/bin/sh -c**).  

**PS**: I have not seen that information in the [reference](https://code.visualstudio.com/docs/remote/containers#_devcontainerjson-reference)

**Comment by Chuxel** [comment]
@chrmarti Yeah, that's a good point.  It's not there. Can add it.  Flipping to doc issue.

**Comment by Chuxel** [comment]
This was resolved with the 1.42 release.
